### PR TITLE
fix: loading relations without DP fails after migrating to v5

### PR DESCRIPTION
**Pull request by Marc-Roig**
### What does it do?

A migration to add the publishedAt column was not setting a default value for content types without DP. 

Resulted in non DP entries to have null values on the publishedAt column. It failed when trying to load relations to such content types with an unexpected NULL publishedAt column. 

This PR:
- Fixes the migration
- Prevents this being an issue at all. Loading relations to nonDP content types do not require filtering by the publishedAt attribute

Fixes https://github.com/strapi/strapi/issues/21166

### CODE REVIEW

File: packages/core/database/src/migrations/internal-migrations/5.0.0-04-published-at.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ import type { Migration } from '../common';
  */
 const createPublishedAtColumn = async (db: Knex, tableName: string) => {
   await db.schema.alterTable(tableName, (table) => {
-    table.string('published_at');
+    table.string('published_at').defaultTo(new Date().getTime());
```
Comment (Marc-Roig) [inline packages/core/database/src/migrations/internal-migrations/5.0.0-04-published-at.ts:14]: I am not completely sure this is how I should format a date, @alexandrebodin  do you remember if we format it differently for each db?

Comment (alexandrebodin) [reply packages/core/database/src/migrations/internal-migrations/5.0.0-04-published-at.ts:14]: you should be ok passing in a new Date()  directly

**Review by alexandrebodin — COMMENTED** [review]
I'm a bit lost, 

If you don't have D&P why should the publishedAt be sth else than null ? 

Setting the defaultTo means everytime we create a row in db the publishedAt will be set to now. Is this really want we want ?

**Comment by Marc-Roig** [comment]
@alexandrebodin maybe I missunderstood the knex api, what I assumed was:

- The migration supposed to add the `publishedAt` for non DP content types was not setting its value to a date and was left to null
- Hence there are users noticing issues with non DP entries with invalid publishedAt values (e.g. folders, files)
- My approach was:
	- To fix the migration by adding the default value, and I assume this default value only applies to the column creation, but not a default value that will forever be used on row creation 
	- Then, for entries with invalid publication states, I just modified the filtering to ignore `publishedAt` if DP is disabled for a content type

**Comment by alexandrebodin** [comment]
> @alexandrebodin maybe I missunderstood the knex api, what I assumed was:
> 
> * The migration supposed to add the `publishedAt` for non DP content types was not setting its value to a date and was left to null

when creating the column if you set defaultTo it will set it forever so not just for the migration. You will have to create the column without defaultTo and then do an updated query instead

> * Hence there are users noticing issues with non DP entries with invalid publishedAt values (e.g. folders, files)
> * My approach was:
>   
>   * To fix the migration by adding the default value, and I assume this default value only applies to the column creation, but not a default value that will forever be used on row creation
>   * Then, for entries with invalid publication states, I just modified the filtering to ignore `publishedAt` if DP is disabled for a content type

**Review by alexandrebodin — APPROVED** [review]

**Pull request merged**